src/ORM/Search/SearchContext.php

```diff
@@ -179,7 +180,34 @@ public function getQuery($searchParams, $sort = false, $limit = false, $existing
                 $filter->setModel($this->modelClass);
                 $filter->setValue($value);
                 if (!$filter->isEmpty()) {
-                    $query = $query->alterDataQuery([$filter, 'apply']);
+                    $modelObj = Injector::inst()->create($this->modelClass);
+                    if(isset($modelObj->searchableFields()[$key]['match_any'])) {
+                        $query = $query->alterDataQuery(function ($dataQuery) use ($modelObj, $key, $value) {
+                            $searchFields = $modelObj->searchableFields()[$key]['match_any'];
+                            $sqlSearchFields = [];
+                            foreach($searchFields as $dottedRelation){
+                                $relation = substr($dottedRelation, 0, strpos($dottedRelation, '.'));
+                                $relations = explode('.', $dottedRelation);
+                                $fieldName = array_pop($relations);
+
+                                // Apply join
```

Comment:
I don't think we need these inline comments explaining the code or the line breaks above them